### ISSUE TITLE
Fails on subversion project with externals

### ISSUE THREAD

**Issue by berhoel** [mgedmin/check-manifest#45]
In my subversion project that uses externals I get an error message:

```
  File "/usr/local/gltools/python/Python-2.7/lib/python2.7/site-packages/check_manifest.py", line 330, in get_vcs_files
    return normalize_names(vcs.get_versioned_files())
  File "/usr/local/gltools/python/Python-2.7/lib/python2.7/site-packages/check_manifest.py", line 310, in get_versioned_files
    for line in output.splitlines()[1:])
  File "/usr/local/gltools/python/Python-2.7/lib/python2.7/site-packages/check_manifest.py", line 310, in <genexpr>
    for line in output.splitlines()[1:])
IndexError: list index out of range
```

Changing line 310 to 

```
                      for line in output.splitlines()[1:] if line.strip())
```

fixes the problem for me.

Kind regards 
Berthold Höllmann

**Comment by mgedmin** [comment]
Ooh, thank you for this interesting use-case.  I'll have to add a test.

**Comment by berhoel** [comment]
A stripped down version of the output from `svn st -vq` lookes like

```
>svn st -vq
                64       64 berhol       .
                64       58 berhol       Makefile
X                                        setup_utils
                64       58 berhol       tox.ini

Performing status on external item at 'setup_utils':
               110      107 berhol       setup_utils
               110      107 berhol       setup_utils/version.py
```

**Comment by mgedmin** [comment]
I now have a failing test case:
```diff
diff --git a/tests.py b/tests.py
index eb499ff..452fd6e 100644
--- a/tests.py
+++ b/tests.py
@@ -914,6 +914,19 @@ class SvnHelper(VCSHelper):
 class TestSvn(VCSMixin, unittest.TestCase):
     vcs = SvnHelper()
 
+    def test_svn_externals(self):
+        from check_manifest import get_vcs_files
+        self.vcs._run('svnadmin', 'create', 'repo2')
+        repo2_url = 'file:///' + os.path.abspath('repo2').replace(os.path.sep, '/')
+        self.vcs._init_vcs()
+        self.vcs._run('svn', 'propset', 'svn:externals', 'ext %s' % repo2_url, '.')
+        self.vcs._run('svn', 'up')
+        self._create_files(['a.txt', 'ext/b.txt'])
+        self.vcs._run('svn', 'add', 'a.txt', 'ext/b.txt')
+        j = os.path.join
+        self.assertEqual(get_vcs_files(),
+                         ['a.txt', j('ext', 'b.txt')])
+
 
 class TestUserInterface(unittest.TestCase):
```

I'm now trying to decide if I should improve the plain-text filtering (what if "Performing status on external..." is a locale-dependent translatable string?) or if I should switch to parsing `svn st --xml`.

**Comment by mgedmin** [comment]
I pushed a fix to master.  @berhoel, can you verify that it works on your repository now?

**Comment by berhoel** [comment]
I can confirm that "check-manifest" is now working with my repository. Thanks.

**Comment by mgedmin** [comment]
I released 0.23 with this fix.
